app/models/transformation_mapping_item.rb

```diff
@@ -3,14 +3,21 @@ class TransformationMappingItem < ApplicationRecord
   belongs_to :source,      :polymorphic => true
   belongs_to :destination, :polymorphic => true
 
-  validates :source_id, :uniqueness => {:scope => [:transformation_mapping_id, :source_type]}
+  validates :source_id, :uniqueness => {:scope => [:transformation_mapping_id, :source_type, :destination_type] }
+  validates :source_type, :inclusion => { :in => %w[EmsCluster Storage Lan] }
```

Comment:
As a casual viewer, I wouldn't know what this array of hardcoded classes is.  How would I know if a new class I create needs to be added to this list?  Can the classes support some feature so you can instead validate the source_type supports it?

Reply:
I don't have a clear answer for classes support question.  Thinking out loud, shouldn't someone creating a new source_type for TransformationMappingItem look at the model and see that the new classes has to be added to this array.  To answer the concern, I'm going to add 
`  # New source_type class must be added here [EmsCluster Storage Lan NewClassName] 
  validates :source_type, :inclusion => { :in => %w[EmsCluster Storage Lan] }`

Reply:
Yeah that's a good point @jrafanie, @thearifismail isn't the set of supported source types up to the type of transformation mapping?  If we add support for going from Openstack to Kubevirt then we'll need to add all the openstack destination_types to the source_types, but that doesn't mean that a CloudVolume is a valid source_type for a VMware to Ovirt transformation.

I'd rather delegate the set of supported types to the transformation_mapping and let it handle what is supported based on the source and destination ems types.

Reply:
Yes moving validations to transformation mapping is the better and correct way but doing so requires a new separate story/bug because it is going to break existing specs.   Since this PR is for datastore/storage validation, I have removed the array with hardcoded class names.  That said, cluster type is already being validated.

Reply:
Removed the array with hardcoded class names